java/amazon-kinesis-producer-sample/src/com/amazonaws/services/kinesis/producer/sample/SampleProducer.java

```diff
@@ -186,21 +202,82 @@ public static String getArgIfPresent(final String[] args, final int index, final
      *  1. The stream name to use (test is default)
      *  2. The region name to use (us-west-1 in default)
      *  3. The duration of the test in seconds, 5 is the default.
+     *  4. The number of records per second to send, 2000 is the default.
+     *  5. The payload size of each record being sent in bytes, 128 is the default.
+     *  6. The max number of connections to configure the KPL with, 1 is the default.
+     *  7. The requestTimeout in milliseconds to configure the KPL with, 60000 is the default.
+     *  8. The bufferTime in milliseconds to configure the KPL with, 2000 is the default.
+     *  9. The threading model to configure the KPL with, PER_REQUEST is the default.
+     *  10. The threadPoolSize to configure the KPL with, 0 is the default.
      */
     public static void main(String[] args) throws Exception {
-        final String streamName = getArgIfPresent(args, 0, STREAM_NAME);
-        final String region = getArgIfPresent(args, 1, REGION);
-        final String secondsToRunString = getArgIfPresent(args, 2, String.valueOf(SECONDS_TO_RUN_DEFAULT));
-        final int secondsToRun = Integer.parseInt(secondsToRunString);
+        int argIndex = 0;
+        final String streamName = getArgIfPresent(args, argIndex++, STREAM_NAME_DEFAULT);
+        final String region = getArgIfPresent(args, argIndex++, REGION_DEFAULT);
+        final int secondsToRun = Integer.parseInt(getArgIfPresent(args, argIndex++, String.valueOf(SECONDS_TO_RUN_DEFAULT)));
```

Comment:
A helper method that extracts the integer parameter by wrapping the Integer.parseInt and getArgIfPresent calls would make the code a lot easier to read.

Reply:
Addressed